src/commands/txCommands/push.js

```diff
@@ -0,0 +1,130 @@
+		resourceInfo.sort(lastUpdateComparator).map(resource => resource['slug'])
+	);
+
+const readParseResource$ = slug =>
+	txlib.readResource$(slug).flatMap(txlib.parsePluckTrns);
+
+const devGitBranch$ = Rx.Observable
+	.bindNodeCallback(child_process.exec)('git rev-parse --abbrev-ref HEAD')
+	.pluck(0)
+	.map(str => str.slice(0, -1));
+
+// Branch whether in dev or CI. Replaces forward slashes because
+// branch names are used as transifex resource names and resource
+// names need to work as valid url paths
+const gitBranch$ = Rx.Observable
+	.if(
+		() => process.env.TRAVIS_PULL_REQUEST_BRANCH,
+		Rx.Observable.of(process.env.TRAVIS_PULL_REQUEST_BRANCH),
+		devGitBranch$
+	)
+	.map(branchname => branchname.replace(/\//g, '_'));
+
+const branchResourceExists$ = Rx.Observable
+	.zip(resources$, gitBranch$)
+	.map(([resources, branch]) => resources.indexOf(branch) > -1);
+
+// syncs content in po format to tx
+const pushResource$ = poData =>
+	Rx.Observable
+		.zip(gitBranch$, branchResourceExists$)
+		.flatMap(([gitBranch, branchResourceExists]) => {
+			if (Object.keys(poData).length) {
+				const pushed$ = branchResourceExists
+					? txlib.updateResource$
+					: txlib.createResource$;
+				return pushed$(gitBranch, poData);
+			} else {
+				return Rx.Observable.if(
+					() => branchResourceExists,
+					txlib.deleteResource$(gitBranch)
+				);
+			}
+		});
+
+// returns 'master' content trn from tx in po format
+const txMasterTrns$ = txlib
+	.readResource$(txlib.MASTER_RESOURCE, txlib.PROJECT_MASTER)
+	.flatMap(txlib.parsePluckTrns);
+
+const resourceTrns$ = Rx.Observable
+	.zip(gitBranch$, resources$)
+	// get resources but filter out my current resource
+	.map(([gitBranch, resources]) =>
+		resources.filter(resource => resource != gitBranch)
+	)
+	// transform resource list to resource content, maintaining order
+	.flatMap(resources =>
+		Rx.Observable.zip.apply({}, resources.map(readParseResource$))
+	)
+	.reduce(
+		(joinedTrns, resourceTrns) => Object.assign(joinedTrns, resourceTrns),
+		{}
+	);
+
+const masterAndResourceTrns$ = Rx.Observable
+	.zip(txMasterTrns$, resourceTrns$)
+	.map(([masterTrns, resourceTrns]) =>
+		Object.assign({}, masterTrns, resourceTrns)
+	);
+
+// don't run against master! will delete trn content on transifex
+gitBranch$.subscribe(branchName => {
+	if (branchName === 'master') {
+		console.log(
+			'do not run this script on master. it will kill the master resource on Transifex.'
+		);
+		process.exit(0);
+	}
+});
+
+const pushContent$ = Rx.Observable
+	.zip(masterAndResourceTrns$, txlib.localTrnsMerged$) // load main and local trn content
+	.do(([main, trns]) =>
+		console.log(
+			'master and resource trns',
+			Object.keys(main).length,
+			' / trns extracted:',
+			Object.keys(trns).length
+		)
+	)
+	.map(txlib.diff) // return local content that is new or updated
+	.do(diff => console.log('trns added / updated:', Object.keys(diff).length))
+	.flatMap(pushResource$)
+
+module.exports = {
```

Comment:
structural nitpick - I think it's more clear to separate the arg-parsing/command-defining code from the 'guts' of the scripts, since the yargs command is just an entry point detail that's independent of how the scripts actually work. Separating the core script logic into `txCommands/util/` or `txCommands/scripts/` will make it easier to separate work on the interface from work on the implementation IMO.